src/coreclr/src/debug/daccess/request.cpp

```diff
@@ -4065,6 +4067,75 @@ PTR_IUnknown ClrDataAccess::DACGetCOMIPFromCCW(PTR_ComCallWrapper pCCW, int vtab
 }
 #endif
 
+#ifdef FEATURE_COMWRAPPERS
+HRESULT ClrDataAccess::DACTryGetComWrappersObjectFromCCW(CLRDATA_ADDRESS ccwPtr, OBJECTREF* objRef)
+{
+    if (ccwPtr == 0 || objRef == NULL)
+        return E_INVALIDARG;
+
+    SOSDacEnter();
+
+    // Read CCWs QI address and compare it to the managed object wrapper's implementation.
+    ULONG32 bytesRead = 0;
+    CLRDATA_ADDRESS vTableAddress = NULL;
+    IfFailGo(m_pTarget->ReadVirtual(ccwPtr, (PBYTE)&vTableAddress, sizeof(CLRDATA_ADDRESS), &bytesRead));
+    if (bytesRead != sizeof(CLRDATA_ADDRESS)
+        || vTableAddress == NULL)
+    {
+        hr = S_FALSE;
```

Comment:
I'd recommend E_FAIL (or E_xxxx). Occasionally we see bugs where people error check functions with SUCCEEDED(foo()) and S_FALSE unintentionally passes that check. I try to shy away from S_FALSE unless there is clear need to distinguish this outcome from other error outcomes.

Reply:
Sorry, this is quite possibly my influence... I use all the `HRESULT` codes :) @jkoritzinsky I think it is me and a few people in Windows who still use `S_FALSE`, we are a dying breed.